src/QueryM/Utxos.purs

```diff
@@ -89,13 +91,12 @@ utxosAt addr = asks _.wallet >>= maybe (allUtxosAt addr) (utxosAtByWallet addr)
   -- by searching "// exclude collateral input from overall utxo set"
   -- or functions getUtxos and checkCollateral.
   namiUtxosAt :: Address -> QueryM (Maybe Transaction.UtxoM)
-  namiUtxosAt address = do
-    utxos' <- allUtxosAt address
-    collateral' <- getWalletCollateral
-    pure do
-      utxos <- unwrap <$> utxos'
-      collateral <- unwrap <$> collateral'
-      pure $ wrap $ Map.delete collateral.input utxos
+  namiUtxosAt address = getWalletCollateral >>= maybe
+    (liftEffect $ throw "Nami wallet missing collateral")
```

Comment:
I'm throwing the error in Aff here, but above it suggests we should move to Either for fine-grained error handling. Although, it seems the errors inside the Maybe are only a small subset of the possible errors from calling Ogmios?

Reply:
I think throwing an exception is fine. This isn't an error that can be meaningfully recovered from. Nami doesn't really work without collateral set

Reply:
That said, we can also change the return type of `utxosAt` to `Either ...`